internal/app/machined/internal/api/reg/reg.go

```diff
@@ -78,7 +78,9 @@ func (r *Registrator) Register(s *grpc.Server) {
 // Reboot implements the machineapi.MachineServer interface.
 func (r *Registrator) Reboot(ctx context.Context, in *empty.Empty) (reply *machineapi.RebootResponse, err error) {
 	reply = &machineapi.RebootResponse{
-		Messages: []*machineapi.Reboot{},
+		Messages: []*machineapi.Reboot{
+			{},
```

Comment:
looks like the problem with broken API response was this thing: we actually returned empty response instead of a single message (as expected).